File: msgraph-mail/java/utilities/src/main/java/graphjavav4/utilities/users/mailFolders/MessageRule.java
package graphjavav4.utilities.users.mailFolders;

import com.microsoft.kiota.serialization.Parsable;
import com.microsoft.kiota.serialization.ParseNode;
import com.microsoft.kiota.serialization.SerializationWriter;
import graphjavav4.utilities.users.Entity;
import java.util.function.BiConsumer;
import java.util.HashMap;
import java.util.Map;
import java.util.Objects;
public class MessageRule extends Entity implements Parsable {
    @javax.annotation.Nullable
    public MessageRuleActions actions;
    @javax.annotation.Nullable
    public MessageRulePredicates conditions;
    @javax.annotation.Nullable
    public String displayName;
    @javax.annotation.Nullable
    public MessageRulePredicates exceptions;
    @javax.annotation.Nullable
    public Boolean hasError;
    @javax.annotation.Nullable
    public Boolean isEnabled;
    @javax.annotation.Nullable
    public Boolean isReadOnly;
    @javax.annotation.Nullable
    public Integer sequence;
    public void serialize(@javax.annotation.Nonnull final SerializationWriter writer) {
        Objects.requireNonNull(writer);
        super.serialize(writer);
        writer.writeObjectValue("actions", actions);
        writer.writeObjectValue("conditions", conditions);
        writer.writeStringValue("displayName", displayName);
        writer.writeObjectValue("exceptions", exceptions);
        writer.writeBooleanValue("hasError", hasError);
        writer.writeBooleanValue("isEnabled", isEnabled);
        writer.writeBooleanValue("isReadOnly", isReadOnly);
        writer.writeIntegerValue("sequence", sequence);
    }
    @javax.annotation.Nonnull
    public <T> Map<String, BiConsumer<T, ParseNode>> getDeserializeFields() {
        final Map<String, BiConsumer<T, ParseNode>> fields = new HashMap<>(super.getDeserializeFields());
        fields.put("actions", (o, n) -> { ((MessageRule)o).actions = n.getObjectValue(MessageRuleActions.class); });
        fields.put("conditions", (o, n) -> { ((MessageRule)o).conditions = n.getObjectValue(MessageRulePredicates.class); });
        fields.put("displayName", (o, n) -> { ((MessageRule)o).displayName = n.getStringValue(); });
        fields.put("exceptions", (o, n) -> { ((MessageRule)o).exceptions = n.getObjectValue(MessageRulePredicates.class); });
        fields.put("hasError", (o, n) -> { ((MessageRule)o).hasError = n.getBooleanValue(); });
        fields.put("isEnabled", (o, n) -> { ((MessageRule)o).isEnabled = n.getBooleanValue(); });
        fields.put("isReadOnly", (o, n) -> { ((MessageRule)o).isReadOnly = n.getBooleanValue(); });
        fields.put("sequence", (o, n) -> { ((MessageRule)o).sequence = n.getIntegerValue(); });
        return fields;
    }
}

File: msgraph-mail/typescript/src/models/messageRule.ts
import { deserializeIntoEntity, serializeEntity, type Entity } from './entity';
import { createMessageRuleActionsFromDiscriminatorValue, serializeMessageRuleActions, type MessageRuleActions } from './messageRuleActions';
import { createMessageRulePredicatesFromDiscriminatorValue, serializeMessageRulePredicates, type MessageRulePredicates } from './messageRulePredicates';
import { type Parsable, type ParseNode, type SerializationWriter } from '@microsoft/kiota-abstractions';

export function createMessageRuleFromDiscriminatorValue(parseNode: ParseNode | undefined) {
    if(!parseNode) throw new Error("parseNode cannot be undefined");
    return deserializeIntoMessageRule;
}
export function deserializeIntoMessageRule(messageRule: MessageRule | undefined = {} as MessageRule) : Record<string, (node: ParseNode) => void> {
    return {
        ...deserializeIntoEntity(messageRule),
        "actions": n => { messageRule.actions = n.getObjectValue<MessageRuleActions>(createMessageRuleActionsFromDiscriminatorValue); },
        "conditions": n => { messageRule.conditions = n.getObjectValue<MessageRulePredicates>(createMessageRulePredicatesFromDiscriminatorValue); },
        "displayName": n => { messageRule.displayName = n.getStringValue(); },
        "exceptions": n => { messageRule.exceptions = n.getObjectValue<MessageRulePredicates>(createMessageRulePredicatesFromDiscriminatorValue); },
        "hasError": n => { messageRule.hasError = n.getBooleanValue(); },
        "isEnabled": n => { messageRule.isEnabled = n.getBooleanValue(); },
        "isReadOnly": n => { messageRule.isReadOnly = n.getBooleanValue(); },
        "sequence": n => { messageRule.sequence = n.getNumberValue(); },
    }
}
export interface MessageRule extends Entity, Parsable {
    /**
     * The actions property
     */
    actions?: MessageRuleActions | undefined;
    /**
     * The conditions property
     */
    conditions?: MessageRulePredicates | undefined;
    /**
     * The display name of the rule.
     */
    displayName?: string | undefined;
    /**
     * The exceptions property
     */
    exceptions?: MessageRulePredicates | undefined;
    /**
     * Indicates whether the rule is in an error condition. Read-only.
     */
    hasError?: boolean | undefined;
    /**
     * Indicates whether the rule is enabled to be applied to messages.
     */
    isEnabled?: boolean | undefined;
    /**
     * Indicates if the rule is read-only and cannot be modified or deleted by the rules REST API.
     */
    isReadOnly?: boolean | undefined;
    /**
     * Indicates the order in which the rule is executed, among other rules.
     */
    sequence?: number | undefined;
}
export function serializeMessageRule(writer: SerializationWriter, messageRule: MessageRule | undefined = {} as MessageRule) : void {
        serializeEntity(writer, messageRule)
        writer.writeObjectValue<MessageRuleActions>("actions", messageRule.actions, );
        writer.writeObjectValue<MessageRulePredicates>("conditions", messageRule.conditions, );
        writer.writeStringValue("displayName", messageRule.displayName);
        writer.writeObjectValue<MessageRulePredicates>("exceptions", messageRule.exceptions, );
        writer.writeBooleanValue("hasError", messageRule.hasError);
        writer.writeBooleanValue("isEnabled", messageRule.isEnabled);
        writer.writeBooleanValue("isReadOnly", messageRule.isReadOnly);
        writer.writeNumberValue("sequence", messageRule.sequence);
}
